File: C#Folder/Account-Polymorphic-App/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Account_Polymorphic_App
{
	abstract class Account
    {
		private int accNo;
		private String accountName;
		protected double balance;
		protected bool transactionStatus = true;

		public Account(int accNo, String accountName, double balance)
		{
			this.accNo = accNo;
			this.accountName = accountName;
			this.balance = balance;
		}

		// getters
		public int getAccNo()
		{
			return accNo;
		}

		public String getAccountName()
		{
			return accountName;
		}

		public double getBalance()
		{
			return balance;
		}

		public bool getTransactionStatus()
		{
			return transactionStatus;
		}

		public void deposit(int amount)
		{
			balance = balance + amount;
		}

		public abstract void withdraw(int amount);

	}
}

File: CoreJava/pattern-proxy-app/src/com/techlab/account/Account.java
package com.techlab.account;

public class Account {
	private String accountName;
	private double balance;
	private String accountNo;

	public Account(String accountNo, String accountName, double balance) {

		this.accountName = accountName;
		this.balance = balance;
		this.accountNo = accountNo;
	}

	public Account(String accountName, double balance) {
		this.accountName = accountName;
		this.balance = balance;
	}

	public void deposit(int amount) throws InterruptedException {
		Thread.sleep(2000);
		balance = balance + amount;
	}

	public void withdraw(int amount) {
		balance = balance - amount;
	}

	public String getAccountName() {
		return accountName;
	}

	public double getBalance() {
		return balance;
	}

	public String getAccountNo() {
		return accountNo;
	}

}